--- a/entityreference_view_widget.module
+++ d/entityreference_view_widget.module
@@ -428,6 +428,15 @@
         $label = entity_label($settings['target_type'], $entities[$entity_id]);
         $suffix = check_plain($label);
       }
+
+      // @see https://wellnet.atlassian.net/browse/CDP-61
+      $flag = flag_get_flag('archived');
+      $node = entity_object_load($entity_id,'node');
+
+      // if flag exist and node is flagged "Archived
+      if ($flag && $flag->is_flagged($node->nid)) {
+        $suffix .= t(' (Archived)');
+      }
       $name = $settings['element'] . '[' . $settings['langcode'] . '][' . $delta . ']';
 
       $rows[] = array(
