diff --git a/includes/service.inc b/includes/service.inc
index 105cdd2..c3fef95 100644
--- a/includes/service.inc
+++ b/includes/service.inc
@@ -1604,8 +1604,15 @@ class SearchApiSolrService extends SearchApiAbstractService {
     if ($value === NULL) {
       return ($operator == '=' ? '*:* AND -' : '') . "$field:[* TO *]";
     }
-    $value = trim($value);
-    $value = $this->formatFilterValue($value, search_api_extract_inner_type($field_info['type']));
+    if (!is_array($value)) {
+      $value = trim($value);
+      $value = $this->formatFilterValue($value, search_api_extract_inner_type($field_info['type']));
+    }
+    else {
+      foreach($value as &$val) {
+        $val = $this->formatFilterValue(trim($val), search_api_extract_inner_type($field_info['type']));;
+      }
+    }
     switch ($operator) {
       case '<>':
         return "*:* AND -($field:$value)";
@@ -1617,6 +1624,10 @@ class SearchApiSolrService extends SearchApiAbstractService {
         return "$field:[$value TO *]";
       case '>':
         return "$field:{{$value} TO *}";
+      case 'between':
+        return "$field:[{$value[0]} TO {$value[1]}]";
+      case 'not between':
+        return "-$field:[{$value[0]} TO {$value[1]}]";
 
       default:
         return "$field:$value";
@@ -1634,7 +1645,7 @@ class SearchApiSolrService extends SearchApiAbstractService {
       case 'date':
         $value = is_numeric($value) ? (int) $value : strtotime($value);
         if ($value === FALSE) {
-          return 0;
+          return '*';
         }
         $value = format_date($value, 'custom', self::SOLR_DATE_FORMAT, 'UTC');
         break;
