diff --git a/sites/all/modules/contrib/inline_entity_form/inline_entity_form.info b/sites/all/modules/contrib/inline_entity_form/inline_entity_form.info
index 951cc91..4317799 100644
--- a/sites/all/modules/contrib/inline_entity_form/inline_entity_form.info
+++ b/sites/all/modules/contrib/inline_entity_form/inline_entity_form.info
@@ -1,6 +1,7 @@
 name = Inline Entity Form
 description = "Provides a widget for inline management (creation, modification, removal) of referenced entities. "
 package = Fields
+configure = admin/config/content/inline-entity-form
 dependencies[] = entity
 dependencies[] = system (>7.14)
 core = 7.x
diff --git a/sites/all/modules/contrib/inline_entity_form/inline_entity_form.module b/sites/all/modules/contrib/inline_entity_form/inline_entity_form.module
index 0e39c5d..160b9de 100644
--- a/sites/all/modules/contrib/inline_entity_form/inline_entity_form.module
+++ b/sites/all/modules/contrib/inline_entity_form/inline_entity_form.module
@@ -29,7 +29,14 @@ function inline_entity_form_menu() {
     'access callback' => TRUE,
     'type' => MENU_CALLBACK,
   );
-
+  $items['admin/config/content/inline-entity-form'] = array(
+    'title' => 'Inline Entity form',
+    'description' => 'Draggable table configuration',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('_inline_entity_form_admin'),
+    'access arguments' => array('administer modules '),
+    'type' => MENU_NORMAL_ITEM,
+  );
   return $items;
 }
 
@@ -1596,7 +1603,7 @@ function theme_inline_entity_form_entity_table($variables) {
   // Sort the fields by weight.
   uasort($fields, 'drupal_sort_weight');
   // If one of the rows is in form context, disable tabledrag.
-  $has_tabledrag = TRUE;
+  $has_tabledrag = variable_get('inline_entity_form_has_draggable', TRUE);
   foreach (element_children($form) as $key) {
     if (!empty($form[$key]['form'])) {
       $has_tabledrag = FALSE;
@@ -1754,3 +1761,30 @@ function inline_entity_form_pre_render_add_fieldset_markup($form) {
 
   return $form;
 }
+
+/**
+ * Custom callback for module settings
+ * @return mixed
+ */
+function _inline_entity_form_admin() {
+  $form = array();
+
+  $form['has_draggable'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Has draggable'),
+    '#default_value' => variable_get('inline_entity_form_has_draggable'),
+    '#description' => t("Active / Deactive draggable table for Inline Entity Form"),
+  );
+  $form['#submit'][] = '_inline_entity_form_admin_submit'; //Submit button call back.
+
+  return system_settings_form($form);
+}
+
+/**
+ * Submit for module settings form
+ * @param $form
+ * @param $form_state
+ */
+function _inline_entity_form_admin_submit(&$form, $form_state) {
+  variable_set('inline_entity_form_has_draggable', $form_state['values']['has_draggable']);
+}
\ No newline at end of file
